common/src/main/java/io/pravega/schemaregistry/common/NameUtil.java

```diff
@@ -59,4 +60,18 @@ public static String qualifiedName(String qualifier, String name) {
         Preconditions.checkNotNull(name, "Name cannot be null");
         return Strings.isNullOrEmpty(qualifier) ? name : String.format("%s.%s", qualifier, name);
     }
+
+    /**
+     * Type value if the 'type' is not null of empty.
+     * If type is null or empty then the created name is simply 'namespace.groupName'.
+     *
+     * @param type the value provided with API call (schemaInfo.getType()).
+     * @param groupName the name of the group for schema
+     * @param namespace the namespace for the schema
+     * @return Provided name or Created name for type in SchemaInfo
+     */
+    public static String createTypeIfAbsent(String type, String groupName, String namespace) {
+        String nameSpace = Strings.isNullOrEmpty(namespace) ? DEFAULT_TYPE : namespace;
```

Comment:
This does not look right to me. Why are we changing the default namespace. This will have consequences on backward compatibility.

Reply:
We are not changing the default namespace, here we're providing a type_name to JSON and Custom schemas if the namespace provided is null or empty. I'll change the variable name, it looks confusing

Reply:
Updated the name of the variable